lib/workers/global/ssh_socket.ts

```diff
@@ -0,0 +1,40 @@
+import type { HostRule } from '../../types';
+import * as cp from 'child_process';
+import * as net from 'net';
+import Agent from 'ssh-agent-js/client/index';
+import { join } from 'upath';
+
+export class SshSocket {
+  agent_socket: string;
+  agent_process: cp.ChildProcess;
+  socket: net.Socket;
+  agent: Agent;
+
+  async start(cacheDir: string) {
```

Comment:
do not start if ssh socket already set in env.

Reply:
Done. Of course this may become problematic when people are running renovate on a host where an ssh-agent is already running but the containers do not have access to the socket 🤔

Reply:
if there is a agent running, but not usable by renovate then it's a user error, not renovate.